### PR TITLE
[On-chain] chore: add single param updates to `shared` module

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Adds a `MsgUpdateParam` message to the `shared` module so that it supports individual param updates.

## Issue

- #517 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/shared/types/message_update_param.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgUpdateParam{}
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply x/shared/types/message_update_param.go:9]:
```suggestion
var _ sdk.Msg = (*MsgUpdateParam)(nil)
```

File: Makefile [hunk 2]
```diff
@@ -833,11 +833,16 @@ params_update_proof_all: ## Update the proof module params
...
 	poktrolld tx authz exec ./tools/scripts/params/proof_min_relay_difficulty_bits.json $(PARAM_FLAGS)
 
-### Session Module Params ###
-.PHONY: params_update_session_all
+### Shared Module Params ###
+.PHONY: params_update_shared_all
 params_update_shared_all: ## Update the session module params
 	poktrolld tx authz exec ./tools/scripts/params/shared_all.json $(PARAM_FLAGS)
 
+### Shared Module Params ###
```
Comment (bryanchriswhite) [inline Makefile:841]:
```suggestion
```

File: x/shared/keeper/msg_server_update_param.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+func (k msgServer) UpdateParam(ctx context.Context, msg *types.MsgUpdateParam) (*types.MsgUpdateParamResponse, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	if k.GetAuthority() != msg.Authority {
+		//return nil, types.ErrSharedInvalidSigner.Wrapf("invalid authority; expected %s, got %s", k.GetAuthority(), msg.Authority)
+		return nil, fmt.Errorf("%s", "...")
```
Comment (bryanchriswhite) [inline x/shared/keeper/msg_server_update_param.go:17]:
```suggestion
		return nil, types.ErrSharedInvalidSigner.Wrapf("invalid authority; expected %s, got %s", k.GetAuthority(), msg.Authority)
```

File: x/shared/keeper/msg_server_update_param_test.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+func TestMsgUpdateParam_UpdateNumBlocksPerSession(t *testing.T) {
+	var expectedNumBlocksPerSession int64 = 8
+
+	k, ctx := keepertest.SharedKeeper(t)
+	msgSrv := keeper.NewMsgServerImpl(k)
+
+	// Set the parameters to their default values
+	//k, msgSrv, ctx := setupMsgServer(t)
```
Comment (bryanchriswhite) [inline x/shared/keeper/msg_server_update_param_test.go:23]:
```suggestion
```

File: Makefile [hunk 5]
```diff
@@ -833,11 +833,15 @@ params_update_proof_all: ## Update the proof module params
...
 
-### Session Module Params ###
-.PHONY: params_update_session_all
+### Shared Module Params ###
+.PHONY: params_update_shared_all
 params_update_shared_all: ## Update the session module params
 	poktrolld tx authz exec ./tools/scripts/params/shared_all.json $(PARAM_FLAGS)
 
+.PHONY: params_update_shared_num_blocks_per_session
+params_update_shared_num_blocks_per_session: ## Update the session module params
```
Comment (bryanchriswhite) [inline Makefile:842]:
```suggestion
params_update_shared_num_blocks_per_session: ## Update the shared module params
```

File: proto/poktroll/shared/tx.proto [hunk 6]
```diff
@@ -12,29 +13,45 @@ option go_package = "github.com/pokt-network/poktroll/x/shared/types";
...
+  Params params = 2 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
 }
 
 // MsgUpdateParamsResponse defines the response structure for executing a
 // MsgUpdateParams message.
-message MsgUpdateParamsResponse {}
\ No newline at end of file
+message MsgUpdateParamsResponse {}
+
+message MsgUpdateParam {
```
Comment (Olshansk) [inline proto/poktroll/shared/tx.proto:40]: Please add the same godoc we have elsewhere

File: x/shared/module/autocli.go [hunk 7]
```diff
@@ -23,11 +23,17 @@ func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
 		Tx: &autocliv1.ServiceCommandDescriptor{
 			Service:              modulev1.Msg_ServiceDesc.ServiceName,
 			EnhanceCustomCommand: true, // only required if you want to use the custom command
-			RpcCommandOptions: []*autocliv1.RpcCommandOptions{
-				{
-					RpcMethod: "UpdateParams",
-					Skip:      true, // skipped because authority gated
-				},
+			RpcCommandOptions:    []*autocliv1.RpcCommandOptions{
+				//	{
```
Comment (Olshansk) [inline x/shared/module/autocli.go:27]: 👀

File: x/shared/types/message_update_param_test.go [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/sample"
+)
+
+func TestMsgUpdateParam_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name        string
```
Comment (Olshansk) [inline x/shared/types/message_update_param_test.go:13]: s/name/desc

**Review by Olshansk — APPROVED** [review]

**Pull request merged**